src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const STATES = [];
+  let copy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        copy = {};
+        break;
+      case 'addProperties':
+        copy = { ...copy, ...action.extraData };
+        break;
+      case 'removeProperties':
+        copy = { ...copy };
+
+        for (const key of action.keysToRemove) {
+          delete copy[key];
+        }
+        break;
+      default:
+        break;
```

Comment:
[CODE STYLE]: It is recommended to have a default case in a switch statement for error handling. Even though you have a default case that breaks, it would be helpful to either throw an error or log a message for an unexpected 'action.type'.